src/pybind/mgr/cephadm/module.py

```diff
@@ -2193,6 +2197,27 @@ def _refresh_configs(self):
                 self._create_daemon(dd.daemon_type, dd.daemon_id,
                                     dd.hostname, reconfig=True)
 
+        # make sure the dashboard [does not] references grafana
+        try:
+            current_url = self.remote('dashboard', 'get_option',
+                                      'GRAFANA_API_URL')
+            current_ssh_verify = self.remote('dashboard', 'get_option',
+                                             'GRAFANA_API_SSL_VERIFY')
+            if grafanas:
+                host = grafanas[0].hostname
+                url = 'https://%s:3000' % (self.inventory[host].get('addr',
+                                                                    host))
+                if current_url != url:
+                    self.log.info('Setting dashboard grafana config to %s' % url)
+                    self.mon_command({
+                        'prefix': 'dashboard set-grafana-api-url',
```

Comment:
Maybe `self.set_module_option_ex` would work here instead?

Reply:
yep, that is much simpler!  also works even when the dashboard module is disabled.